CenterForOpenScience/osf.io
# [ENG-3110] NotableEmailDomain (and a bit of domain-based trust)

## Purpose
Avoid effort wasted dealing with automated spam-check false positives.


## Changes
* rename `BlacklistedEmailDomain` to `NotableEmailDomain`
* add `NotableEmailDomain.note` field with two possible values:
  * `EXCLUDE_FROM_ACCOUNT_CREATION`: existing behavior of `BlacklistedEmailDomain`, prevent creating accounts with emails from this domain
  * `ASSUME_NONSPAM_UNTIL_REPORTED`: (aka "trusted") if a user has an email from this domain, skip spam checks on content they post
* allow bulk-adding NotableEmailDomains in the admin interface



## QA Notes
gonna dev-test

Please make verification statements inspired by your code and what your code touches.
- Verify spam checks run for users without a trusted email domain
- Verify spam checks don't run for users with a trusted email domain

What are the areas of risk?
- touched logic for existing "exclude from account creation" domains -- should still work as before

Any concerns/considerations/questions that development raised?

## Documentation
- internal support docs for osf-admin:
  - meaning/effects of NotableEmailDomain's two notes
  - how to view/search/delete/edit/add/bulk-add NotableEmailDomains



## Side Effects
n/a


## Ticket
[ENG-3110]



[ENG-3110]: https://openscience.atlassian.net/browse/ENG-3110?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
